### ISSUE TITLE
Hydration errors when using `SSRMountStyleProvider` with Suspense/Transition during SSR

### ISSUE THREAD

**Issue by kiperz** [thaw-ui/thaw#346]
I've encountered hydration errors when using the component introduced in [#335](https://github.com/thaw-ui/thaw/pull/335) while implementing `Suspense`/`Transition` with Server-Side Rendering (SSR).

Below is a minimal reproducible example demonstrating the issue:

```rust
use leptos::prelude::*;
use leptos_meta::{MetaTags, provide_meta_context};
use thaw::ssr::SSRMountStyleProvider;
use thaw::{ConfigDirection, ConfigProvider, Spinner};

pub fn shell(options: LeptosOptions) -> impl IntoView {
    view! {
        <SSRMountStyleProvider>
            <!DOCTYPE html>
            <html lang="en">
                <head>
                    <meta charset="utf-8" />
                    <meta name="viewport" content="width=device-width, initial-scale=1" />
                    <AutoReload options=options.clone() />
                    <HydrationScripts options />
                    <MetaTags />
                </head>
                <body>
                    <App />
                </body>
            </html>
        </SSRMountStyleProvider>
    }
}

async fn fetch_cats(_: u32) -> Vec<String> {
    vec![]
}

#[component]
pub fn App() -> impl IntoView {
    provide_meta_context();
    let dir = RwSignal::new(ConfigDirection::Ltr);
    let (cat_count, _) = signal::<u32>(1);

    let cats = Resource::new(move || cat_count.get(), |count| fetch_cats(count));

    view! {
        <ConfigProvider dir=dir>
            <Transition fallback=move || {
                view! { <Spinner /> }
            }>
                {move || Suspend::new(async move {
                    cats.await
                        .into_iter()
                        .map(|src| {
                            view! { <img src=src /> }
                        })
                        .collect_view()
                })}
            </Transition>
        </ConfigProvider>
    }
}
```

I'm currently using the latest versions of both Thaw (`0.4.0`) and Leptos (`0.7.2`).

**Comment by luoxiaozero** [comment]
I fixed this issue in #347. Can you try it in your project?

```
thaw = { git = "https://github.com/thaw-ui/thaw", rev = "cfbcad1" }
```

**Comment by kiperz** [comment]
Thank you for promptly addressing the initial issue; the fix appears to resolve the hydration errors successfully. However, I've encountered another concern. Upon page load, there is a noticeable period where the page renders without the applied, resulting in a flash of unstyled content(FOUC). This behavior suggests that styles are being inserted only after client-side hydration, rather than included during server-side rendering.

Ideally, it should be possible to determine the components that will be loaded at application that the can be and included during the initial server-side render This would prevent FOUC and ensure user experience without discrepancies during the load.

**Comment by kiperz** [comment]
I have found yet another issue, SSR style rendering is appending ConfigProvider style tag, but after hydration this style tag is being duplicated by hydration process.

**Comment by luoxiaozero** [comment]
The problem comes from the fact that hydration and SSR cannot generate the same id. I added a theme_id parameter to the ConfigProvider component for users to specify an id themselves.

**Comment by kiperz** [comment]
It appears to be functioning correctly now, thank you once again for the prompt resolution. I am pleased to report that it works as intended although it still briefly displays unstyled content for approximately 100ms. It seems that the hydrated styles replace generated before sending the HTML to client.
